qubes-dist-upgrade.sh

```diff
@@ -719,108 +324,53 @@ if [ "$assumeyes" == "1" ] || confirm "-> Launch upgrade process?"; then
-                systemctl preset qubes-qrexec-policy-daemon
-                systemctl preset logrotate systemd-pstore
-
-                # this file is created by salt on fresh install if sys-usb is
-                # enabled, do it here on upgrade
-                if systemctl -q is-enabled "qubes-vm@$usbvm.service"; then
-                    mkdir -p "/etc/systemd/system/qubes-vm@$usbvm.service.d"
-                    cat >"/etc/systemd/system/qubes-vm@$usbvm.service.d/50_autostart.conf" <<EOF
-[Unit]
-Before=systemd-user-sessions.service
-EOF
-                fi
-
-                # Update legacy Grub if needed
-                update_legacy_grub
-            else
-                echo "WARNING: dist-upgrade stage canceled."
-            fi
-        else
-            false
-        fi
-    fi
-
-    if [ "$update_grub" == "1" ]; then
-        echo "---> (STAGE 5) Installing EFI Grub..."
-        setup_efi_grub
-    fi
-    echo "INFO: Please ensure to have completed all the stages and reboot before continuing."
+    # Executing post upgrade tasks
+  if [ "$finalize" == 1 ]; then
+      echo "---> (STAGE 5) Synchronizing menu entries and supported features"
+      if [ "$skip_template_upgrade" != 1 ]; then
+          mapfile -t template_vms < <(qvm-ls --raw-data --fields name,klass | grep 'TemplateVM$' | cut -d '|' -f 1)
+      fi
+      if [ "$skip_standalone_upgrade" != 1 ]; then
+          mapfile -t standalone_vms < <(qvm-ls --raw-data --fields name,klass | grep 'StandaloneVM$' | cut -d '|' -f 1)
+      fi
+      if [ "$skip_template_upgrade" != 1 ] || [ "$skip_standalone_upgrade" != 1 ]; then
+          mapfile -t all_vms < <(echo "${template_vms[@]}" "${standalone_vms[@]}")
+      fi
+      if [ -n "$only_update" ]; then
+          IFS=, read -ra all_vms <<<"${only_update}"
+      fi
+      if [ "${#all_vms[*]}" -gt 0 ]; then
+          for vm in ${all_vms[*]};
+          do
+              if ! qvm-run --service "$vm" qubes.PostInstall; then
+                  echo "WARNING: Failed to execute qubes.PostInstall in $vm."
+              fi
+              qvm-shutdown "$vm"
+          done
+      fi
+      user=$(groupmems -l -g qubes | cut -f 1 -d ' ')
+      runuser -u "$user" -- qvm-appmenus --all --update
+
+      echo "---> (STAGE 5) Cleaning up salt"
+      echo "Error on ext_pillar interface qvm_prefs is expected"
+      qubesctl saltutil.clear_cache
+      qubesctl saltutil.sync_all
+
+      echo "---> (STAGE 5) Adjusting default kernel"
+      default_kernel="$(qubes-prefs default-kernel)"
+      default_kernel_path="/var/lib/qubes/vm-kernels/$default_kernel"
+      default_kernel_package="$(rpm --qf '%{NAME}' -qf "$default_kernel_path")"
+      if [ "$default_kernel_package" = "kernel-qubes-vm" ]; then
+          new_kernel=$(rpm -q --qf '%{VERSION}-%{RELEASE}\n'  kernel-qubes-vm | sort -V | tail -1)
+          new_kernel="${new_kernel%.qubes}"  # TODO: does this work? check with tests
```

Comment:
No, it doesn't, there may be `6.1.38-1.qubes.fc37`. Try `new_kernel="${new_kernel/.qubes/}"`